proposals/0173-Read-Generic-Network-Signal-data.md

```diff
@@ -16,277 +16,306 @@ To enable OEM proprietary mobile apps to read raw CAN/Network data by specifying
+	</param>
+	<param name="ambientLightSensorStatus" type="AmbientLightStatus" mandatory="false">
+		<description>Status of the ambient light sensor.</description>
+	</param>
+</struct>
 
-    <param name="gps" type="Boolean" mandatory="false">
-      <description>See GPSData</description>
-    </param>
     
-	<!-- Not showing other params -->
-	
-	<param name="networkDataRequest" type="NetworkDataRequest" mandatory="false">	
-      <description>Request Generic Network data</description>
-    </param>
-</function>
-  
-  
-<struct name="NetworkDataRequest">
-	<param name="paramId" type="Integer" minvalue="1" maxvalue="65535" mandatory="true">
-      <description>Paramter ID for the request.</description>
-    </param>
-	
-	<param name="NetworkDataAttribute" type="String" maxlength="500" minsize="1" maxsize="100" array="true" mandatory="true">
-      <description>Array of Network Data Attributes which uniquely identify network data as per OEM implementation 
-	  Should be uniquely identifiable network data item, else response would be INVALID_DATA
-	  </description>
-    </param>
-</struct>
-```
+<enum name="AmbientLightStatus">
+	<description>Reflects the status of the ambient light sensor.</description>
+	<element name="NIGHT" />
+	<element name="TWILIGHT_1" />
+	<element name="TWILIGHT_2" />
+	<element name="TWILIGHT_3" />
+	<element name="TWILIGHT_4" />
+	<element name="DAY" />
+	<element name="UNKNOWN" internal_name="ALS_UNKNOWN" />
+	<element name="INVALID" />
+</enum> 
 
-* For GetVehicleData response/OnVehicleData, add additional response structure:
-  * networkData - It has a paramId and value returned from vehicle. Value returned will be a Hex string.
-  
 ```
-<function name="GetVehicleData" functionID="GetVehicleDataID" messagetype="response">
 
-    <param name="success" type="Boolean" platform="documentation">
-		<description> true, if successful; false, if failed </description>
-	</param>
-
-    <param name="resultCode" type="Result" platform="documentation">
-      <description>See Result</description>
-      <element name="SUCCESS"/>
-      <element name="INVALID_DATA"/>
-      <element name="OUT_OF_MEMORY"/>
-      <element name="TOO_MANY_PENDING_REQUESTS"/>
-      <element name="APPLICATION_NOT_REGISTERED"/>
-      <element name="GENERIC_ERROR"/>
-      <element name="REJECTED"/>
-      <element name="VEHICLE_DATA_NOT_AVAILABLE"/>
-      <element name="USER_DISALLOWED"/>
-    </param>
+* As you see, there are 3 sub items, “ambientLightSensorStatus”, “highBeamsOn” and “lowBeamsOn”. Each of these individual data types and references
+* You many notice that “ambientLightSensorStatus” has data type as “AmbientLightStatus”. This enumeration is read from existing Mobile API.
+  * This can be extrapolated to all existing data types. For new data types enumerations, we’d still either need to add to Mobile API or create a new structure for vehicle data enumerations.
```

Comment:
Change to:
This can be extrapolated to all existing data types. **For new data types enumerations, a Mobile API or a new created structure for vehicle data enumerations would need to be added.**

Reply:
Update this please